relayer/chain.go

```diff
@@ -222,12 +223,18 @@ func (src *Chain) BuildAndSignTx(datagram []sdk.Msg) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-
-	return auth.NewTxBuilder(
+	ctx := sdkCtx.CLIContext{Client: src.Client}
```

Comment:
Any reason not to set `Simulate: true`. However, I don't think it should make a difference since this [func](https://github.com/cosmos/cosmos-sdk/blob/d7677e087117fc9d79c590ee511f1f46a8cbd586/x/auth/client/tx.go#L60) isn't being called.

Reply:
Exactly usually the cli uses `CompleteAndBroadcastTxCLI`  where the Simulate value  is being used, but it only prints out the result, I extracted the bits that does the simulation (`EnrichWithGas`) so that we only pass the txs with proper gas estimate, and that the chain own send method  can handle a proper `sdk.TxResponse`